src/k8s/pkg/component/component_test.go

```diff
@@ -0,0 +1,243 @@
+		Capabilities:   chartutil.DefaultCapabilities,
+		RegistryClient: registryClient,
+		Log: func(format string, v ...interface{}) {
+			t.Helper()
+			if *verbose {
+				t.Logf(format, v...)
+			}
+		},
+	}
+}
+
+type chartOptions struct {
+	*chart.Chart
+}
+
+type chartOption func(*chartOptions)
+
+func buildChart(opts ...chartOption) *chart.Chart {
+	c := &chartOptions{
+		Chart: &chart.Chart{
+			Metadata: &chart.Metadata{
+				APIVersion: "v1",
+				Name:       "hello",
+				Version:    "0.1.0",
+			},
+		},
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c.Chart
+}
+
+func namedReleaseStub(name string, status release.Status) *release.Release {
+	now := time.Now()
+	return &release.Release{
+		Name: name,
+		Info: &release.Info{
+			FirstDeployed: now,
+			LastDeployed:  now,
+			Status:        status,
+			Description:   "Named Release Stub",
+		},
+		Config:  map[string]interface{}{"name": "value"},
+		Version: 1,
+	}
+}
+
+// "Mock" Initializer
+type MockHelmClientInitializer struct {
+	actionConfig *action.Configuration
+}
+
+func (r *MockHelmClientInitializer) InitializeHelmClientConfig() (*action.Configuration, error) {
+	return r.actionConfig, nil
+}
+
+func makeMeSomeReleases(store *storage.Storage, t *testing.T) {
+	t.Helper()
+	relStub1 := namedReleaseStub("whiskas-1", release.StatusDeployed)
+	relStub2 := namedReleaseStub("whiskas-2", release.StatusDeployed)
+	relStub3 := namedReleaseStub("whiskas-3", release.StatusDeployed)
+
+	for _, rel := range []*release.Release{relStub1, relStub2, relStub3} {
+		if err := store.Create(rel); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	all, err := store.ListReleases()
+	assert.NoError(t, err)
+	assert.Len(t, all, 3, "sanity test: three items added")
+}
+
+var componentsNone = ``
+
+var components = `
+one:
+  release: "whiskas-1"
+  chart: "chunky-tuna-1.14.1.tgz"
+  namespace: "default"
+two:
+  release: "whiskas-2"
+  chart: "tuna-1.29.0.tgz"
+  namespace: "default"
+three:
+  release: "whiskas-3"
+  chart: "chunky-1.29.0.tgz"
+  namespace: "default"
+`
+
+func createTemporaryTestDirectory(t *testing.T) string {
+	// Create a temporary test directory to mock the snap
+	// <tempDir>
+	// └── k8s/components
+	// 	├── charts
+	// 	└── component.yaml
+	tempDir, err := os.MkdirTemp("", "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	k8sComponentsDir := filepath.Join(tempDir, "k8s", "components")
+	err = os.MkdirAll(k8sComponentsDir, 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	k8sComponentsChartsDir := filepath.Join(k8sComponentsDir, "charts")
+	err = os.MkdirAll(k8sComponentsChartsDir, 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return tempDir
+}
+
+func addConfigToTestDir(t *testing.T, path string, data string) {
+	// Create a file and add some configs
+	err := os.WriteFile(path, []byte(data), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func createNewManager(t *testing.T, components string) (*helmClient, string, *action.Configuration) {
```

Comment:
nit: Go prefers `mustxxx` for setup steps that fatally fail, e.g. `mustCreateNewManager` and `mustCreateTestDir`